### PR TITLE
Add max length to report name and enforce it on clone [PD-2370]

### PR DESCRIPTION
**Pull request by tdphillips**
I verified this in all window widths that I could, including down to my browser's minimum of 475px. I'm open to suggestions if that is not adequate.

### CODE REVIEW

File: gulp/src/reporting/SetUpReport.jsx [hunk 1]
```diff
@@ -338,4 +341,9 @@ SetUpReport.propTypes = {
       reportDataId: PropTypes.string,
     }).isRequired,
   }).isRequired,
+  maxNameLength: PropTypes.number,
+};
+
+SetUpReport.defaultProps = {
+  maxNameLength: 24,
```
Comment (tdphillips) [inline gulp/src/reporting/SetUpReport.jsx]: Because magic numbers are :hankey:

File: gulp/src/reporting/SetUpReport.jsx [hunk 2]
```diff
@@ -79,7 +79,8 @@ export default class SetUpReport extends Component {
   }
 
   onReportNameChanged(reportName) {
-    this.setState({reportName});
+    const {maxNameLength} = this.props;
+    this.setState({reportName: reportName.substring(0, maxNameLength)});
```
Comment (tdphillips) [inline gulp/src/reporting/SetUpReport.jsx]: Ugly? Bad? Annoying? Thoughts?

Comment (misterjonscott) [inline gulp/src/reporting/SetUpReport.jsx]: So we're just going to clip it if it's too long?

**Comment by misterjonscott** [comment]
@tdphillips Looks good, but I'll wait to hear back about the question above.

File: gulp/src/reporting/SetUpReport.jsx (hunk 2 above)
Comment (tdphillips) [inline gulp/src/reporting/SetUpReport.jsx]: I think so.

[Here's Jason's comment on the matter.](https://directemployersfoundation.atlassian.net/browse/PD-2370?focusedCommentId=218838&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-218838) Let me know if you disagree.

**Pull request merged**